File: spec/python/test_term_bytes.py
import unittest

from term_bytes import TermBytes

class TestTermBytes(unittest.TestCase):
    def test_term_bytes(self):
        r = TermBytes.from_file("src/term_strz.bin")

        self.assertEqual(r.s1, b"foo")
        self.assertEqual(r.s2, b"bar")
        self.assertEqual(r.s3, b"|baz@")

File: spec/lua/test_term_bytes.lua
local luaunit = require("luaunit")

require("term_bytes")

TestTermBytes = {}

function TestTermBytes:test_term_bytes()
    local r = TermBytes:from_file("src/term_strz.bin")

    luaunit.assertEquals(r.s1, "foo")
    luaunit.assertEquals(r.s2, "bar")
    luaunit.assertEquals(r.s3, "|baz@")
end
